### PR TITLE
Implement multi tag functionality in the dxTagBox

### PR DESCRIPTION
**Pull request by mrjohn0011**

### CODE REVIEW

File: js/localization/messages/en.json [hunk 1]
```diff
@@ -232,6 +232,9 @@
         "dxColorView-ariaAlpha": "Transparency",
         "dxColorView-ariaHex": "Color code",
 
+        "dxTagBox-selected": "selected",
```
Comment (vconst) [inline js/localization/messages/en.json]: Why is not "{0} selected"? In the other languages, count parameter may be placed at the end, for example.

File: js/localization/messages/en.json [hunk 2]
```diff
@@ -232,6 +232,9 @@
...
+        "dxTagBox-all-selected": "All selected",
```
Comment (vconst) [inline js/localization/messages/en.json]: Why is not "All selected ({0})"?

File: js/localization/messages/ja.json [hunk 3]
```diff
@@ -202,6 +202,8 @@
         "dxColorView-ariaBlue": "青",
         "dxColorView-ariaAlpha": "透明度",
         "dxColorView-ariaHex": "色コード",
+        "dxTagBox-selected": "",
```
Comment (vconst) [inline js/localization/messages/ja.json]: if(otherLocalValue === "!TODO!") {
                assert.ok(true, name + " is marked for translation");
            } else {
                assert.ok(true, name + " is translated");
            }

File: js/ui/tag_box.js [hunk 4]
```diff
@@ -315,6 +315,25 @@ var TagBox = SelectBox.inherit({
...
+             */
+            maxTagCount: undefined,
+
+            /**
+             * @name dxTagBoxOptions_onMultiTagPreparing
+             * @publicName onMultiTagPreparing
+             * @extends Action
+             * @type_function_param1_field4 multiTagElement:jQuery
+             * @type_function_param1_field5 selectedItems:array
+             * @type_function_param1_field6 allSelected:boolean
```
Comment (vconst) [inline js/ui/tag_box.js:339]: Where is cancel parameter?

Comment (vconst) [reply js/ui/tag_box.js:339]: Where is text parameter?

**Review by vconst — CHANGES_REQUESTED** [review]

File: js/ui/tag_box.js [hunk 5]
```diff
@@ -439,6 +468,35 @@ var TagBox = SelectBox.inherit({
...
 
+    _initActions: function() {
+        this.callBase();
+        this._initMultiTagPreparingAction();
+    },
+
+    _initMultiTagPreparingAction: function() {
+        var onMultiTagPreparing = this.option("onMultiTagPreparing");
+
+        if(onMultiTagPreparing) {
```
Comment (lukyanovas) [inline js/ui/tag_box.js]: use afterExecute(dxAction) functionality instead

**Review by lukyanovas — CHANGES_REQUESTED** [review]

**Review by lukyanovas — DISMISSED** [review]

**Review by vconst — APPROVED** [review]

File: js/localization/messages/ja.json [hunk 6]
```diff
@@ -202,6 +202,9 @@
         "dxColorView-ariaBlue": "青",
         "dxColorView-ariaAlpha": "透明度",
         "dxColorView-ariaHex": "色コード",
+        "dxTagBox-selected": "!TODO!",
+        "dxTagBox-all-selected": "!TODO!",
```
Comment (vconst) [inline js/localization/messages/ja.json]: dxTagBox-all-selected -> dxTagBox-allSelected

File: js/localization/messages/ja.json [hunk 7]
```diff
@@ -202,6 +202,9 @@
...
+        "dxTagBox-more-selected": "!TODO!",
```
Comment (vconst) [inline js/localization/messages/ja.json]: dxTagBox-more-selected -> dxTagBox-moreSelected

**Review by vconst — CHANGES_REQUESTED** [review]

**Review by RomanTsukanov — DISMISSED** [review]

**Review by vconst — DISMISSED** [review]

**Review by lukyanovas — DISMISSED** [review]

**Review by vconst — APPROVED** [review]

**Pull request merged**